app/services/chatbot/chatbot.ts

```diff
@@ -0,0 +1,203 @@
+    const headers = authorizedHeaders(this.state.api_token);
+    const request = new Request(url, { headers });
+
+    return fetch(request)
+      .then(handleErrors)
+      .then(response => response.json());
+  }
+
+  fetchTimers(page: number) {
+    const url = this.apiEndpoint(`timers?page=${page}`, true);
+    const headers = authorizedHeaders(this.state.api_token);
+    const request = new Request(url, { headers });
+
+    return fetch(request)
+      .then(handleErrors)
+      .then(response => response.json());
+  }
+
+  //
+  // POST, PUT requests
+  //
+  updateDefaultCommand(slugName: string, commandName: string, data: DefaultCommand) {
+    const url = this.apiEndpoint(`settings/${slugName}/commands/${commandName}`, true);
+    const headers = authorizedHeaders(this.state.api_token);
+    headers.append('Content-Type', 'application/json');
+    const request = new Request(url, {
+      headers,
+      method: 'POST',
+      body: JSON.stringify(data)
+    });
+
+    return fetch(request)
+      .then(handleErrors)
+      .then(response => response.json());
+  }
+
+  updateCustomCommand(id: string, data: CustomCommand) {
+    const url = this.apiEndpoint(`commands/${id}`, true);
+    const headers = authorizedHeaders(this.state.api_token);
+    headers.append('Content-Type', 'application/json');
+    const request = new Request(url, {
+      headers,
+      method: 'PUT',
+      body: JSON.stringify(data)
+    });
+
+    return fetch(request)
+      .then(handleErrors)
+      .then(response => response.json());
+  }
+
+  createCustomCommand(data: CustomCommand) {
+    const url = this.apiEndpoint('commands', true);
+    const headers = authorizedHeaders(this.state.api_token);
+    headers.append('Content-Type', 'application/json');
+    const request = new Request(url, {
+      headers,
+      method: 'POST',
+      body: JSON.stringify(data)
+    });
+
+    return fetch(request)
+      .then(handleErrors)
+      .then(response => response.json());
+  }
+
+  createTimer(data: Timer) {
+    const url = this.apiEndpoint('timers', true);
+    const headers = authorizedHeaders(this.state.api_token);
+    headers.append('Content-Type', 'application/json');
+    const request = new Request(url, {
+      headers,
+      method: 'POST',
+      body: JSON.stringify(data)
+    });
+
+    return fetch(request)
+      .then(handleErrors)
+      .then(response => response.json());
+  }
+
+  updateTimer(id: string, data: Timer) {
+    const url = this.apiEndpoint(`timers/${id}`, true);
+    const headers = authorizedHeaders(this.state.api_token);
+    headers.append('Content-Type', 'application/json');
+    const request = new Request(url, {
+      headers,
+      method: 'PUT',
+      body: JSON.stringify(data)
+    });
+
+    return fetch(request)
+      .then(handleErrors)
+      .then(response => response.json());
+  }
+
+  //
+  // Mutations
+  //
+  @mutation()
+  LOGIN(response: ChatbotApiServiceState) {
```

Comment:
We've started making mutations `private` just to discourage calling them outside the service that defines them.